examples/gfx/opengl/5_opengl_flappy_bird/main.cpp

```diff
@@ -27,7 +27,15 @@ int main() {
     glEnable(GL_BLEND);
     glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
 
-    auto input = window->get_input();
+    auto input = std::make_shared<Input>();
+    auto input_executor = std::make_shared<InputExecutor>(input);
+
+    KeyListener key_listener = KeyListener(input_executor);
+    MouseListener mouse_listener = MouseListener(input_executor);

```

Comment:
same as above

Reply:
That's my code style. I don't think it harms our code in any way. For me, the auto keyword reduces readability. I don't like to use it, except when there is std::shared_pointer.